pkg/receive/multitsdb.go

```diff
@@ -576,6 +604,14 @@ func (t *MultiTSDB) TenantAppendable(tenantID string) (Appendable, error) {
 	return tenant.readyStorage(), nil
 }
 
+func (t *MultiTSDB) SetHashringConfig(cfg []HashringConfig) {
+	t.hashringConfigs = cfg
```

Comment:
Don't we need to update labels here as well?

Reply:
I don't think so. `ExternalLabels` is a field of `HashringConfig`. When we set `HashringConfig` we automatically set `ExternalLabels`.

```
type HashringConfig struct {
	Hashring       string   `json:"hashring,omitempty"`
	Tenants        []string `json:"tenants,omitempty"`
	Endpoints      []string `json:"endpoints"`
	ExternalLabels []string `json:"external_labels"`
}

```
Also, I checked locally with added logs. External labels are successfully automatically updated with hashring configs. The changes just don't show up in query.

Reply:
Right, but here you are changing the hashring config and replacing it with another one. After changing the config, the labels that are returned are still the ones captured from the initial hashring here: https://github.com/thanos-io/thanos/pull/5777/files#diff-61eda56d07cd51eb9fde89c6bd9f1fe7e260edb05eb8fb4c1c81aa258f034331R500

So I suggest you update the labels when the config is set in this method.

Reply:
Oh, you mean updating the label set for each tenant in the `MultiTSDB`?

Reply:
Yes, we can do that here.

Reply:
Ok, I'll try that. But I have a question here:

I thought when hashring changes, the dbs will [flush and open again](https://github.com/thanos-io/thanos/blob/83595175029cfc993f4ba3d3a99de4b6a0996911/cmd/thanos/receive.go#L604-L608). And when it [opens](https://github.com/thanos-io/thanos/blob/83595175029cfc993f4ba3d3a99de4b6a0996911/pkg/receive/multitsdb.go#L192), it will call [getOrLoadTenant](https://github.com/thanos-io/thanos/blob/83595175029cfc993f4ba3d3a99de4b6a0996911/pkg/receive/multitsdb.go#L210), which then calls [startTSDB](https://github.com/thanos-io/thanos/blob/83595175029cfc993f4ba3d3a99de4b6a0996911/pkg/receive/multitsdb.go#L568) again. With that flow, the label set for each tenant is automatically updated.

Is that not correct?

Reply:
startTSDB is only called when a tenant does not exist: https://github.com/thanos-io/thanos/blob/83595175029cfc993f4ba3d3a99de4b6a0996911/pkg/receive/multitsdb.go#L542

Reply:
ah, missed it, got it, thanks!

Reply:
Hey I understand the cases when we add a new tenant, and when we change the label set of an existing tenant in the hashring config. But what if we delete a tenant from the hashring config?